nexus/src/app/rack.rs

```diff
@@ -107,14 +110,47 @@ impl super::Nexus {
         // internally ignores ObjectAlreadyExists, so will not error on repeat runs
         let _ = self.populate_mock_system_updates(&opctx).await?;
 
+        let dns_zone = request
+            .internal_dns_zone_config
+            .zones
+            .into_iter()
+            .find(|z| z.zone_name == internal_dns::DNS_ZONE)
+            .ok_or_else(|| {
+                Error::invalid_request(
+                    "expected initial DNS config to include control plane zone",
+                )
+            })?;
+
+        let internal_dns = InitialDnsGroup::new(
+            DnsGroup::Internal,
+            &dns_zone.zone_name,
+            &self.id.to_string(),
+            "rack setup",
+            dns_zone.records,
+        );
+
+        // TODO the initial external DNS zone name and potentially record
+        // name(s) need to come in with the rack initialization request.
```

Comment:
In terms of making sure this work is covered by _something_ on the MVP board, it's part of #2237.  That issue doesn't have any fine-grained detail about the project, though.  If it would be useful, I can flesh that issue out with either a checklist or file more issues and link them to that ticket.

Reply:
Yeah, just want to make sure the appropriate follow-ups are tagged. 

This seems like data we might want to propagate through https://github.com/oxidecomputer/omicron/blob/af53f40341f13dc265df273d304ce1ae774a6e97/sled-agent/src/bootstrap/params.rs#L17-L40